src/app/server/EchoHandler.cpp

```diff
@@ -40,7 +40,7 @@ chip::Protocols::Echo::EchoServer gEchoServer;
  */
 void HandleEchoRequestReceived(chip::Messaging::ExchangeContext * ec, chip::System::PacketBufferHandle && payload)
 {
-    ChipLogProgress(AppServer, "Echo Request, len=%u ... sending response.\n", payload->DataLength());
+    ChipLogProgress(AppServer, "Echo Request, len=%u ... sending response.\n", static_cast<uint16_t>(payload->DataLength()));
```

Comment:
Converting to uint16_t could drop the value.

For maximum portability and to ensure correctness, you should use the format specifiers defined in <inttypes.h> for fixed-width integers. For uint32_t, the correct format specifier is PRIu32:

ChipLogProgress(AppServer, "Echo Request, len= %" PRIu32 "... sending response.\n", static_cast<uint16_t>(payload->DataLength()));

Reply:
Will change to PRIu32.